plugins/europeana/entity.js

```diff
@@ -299,3 +299,26 @@ export function entityParamsFromUri(uri) {
   const type = getEntityTypeHumanReadable(matched[1]);
   return { id, type };
 }
+
+/**
+ * Return all entity subjects of type concept / agent
+ * @param {Object} params additional parameters sent to the API
+ */
+export function getEntityIndex(params = {}) {
```

Comment:
This duplicates much of `searchEntities`. Refactor so that:
* The function currently named `searchEntitites` is named `findEntities`, updating any existing references to it
* The function currently named `getEntityIndex` is named `searchEntitites`
* `findEntities` calls `searchEntities` with the entity URIs gathered into the `query` property of the `params` argument
* `findEntities` returns just the entities from the response of `searchEntities`, not the total

Reply:
@rwd I have refactored findEntities and SearchEntities, findEntities is working on item pages, but there’s one issue

SearchEntities requires a type as one of it’s params if I view http://localhost:3000/en/item/2048001/Athena_Plus_ProvidedCHO_KIK_IRPA__Brussels__Belgium__AP_10325114
I need to send `type: 'person'` as the param

however, viewing http://localhost:3000/en/item/2048210/europeana_fashion_2005_6_0396 
I need to send `type: 'topic'` as the param

on the collections page we can get this info from `$route.params.type`
on items pages this is not possible since it's not included in the route params

checking the method `getRecord` on the item page and logging what data is returned from result.record I can see that the record api does return a type field, however, its value is ‘IMAGE’

Reply:
@alansutherland try this:
```diff
diff --git a/pages/collections/_type/index.vue b/pages/collections/_type/index.vue
index a95deca2..5b3bf05b 100644
--- a/pages/collections/_type/index.vue
+++ b/pages/collections/_type/index.vue
@@ -43,6 +43,7 @@
   import {
     searchEntities,
     getEntitySlug,
+    getEntityTypeApi,
     getEntityTypeHumanReadable
   } from '../../../plugins/europeana/entity';
 
@@ -67,7 +68,7 @@
       const entityIndexParams = {
         query: '*:*',
         page: currentPage - 1,
-        type: params.type.slice(0, -1),
+        type: getEntityTypeApi(params.type.slice(0, -1)),
         pageSize: PER_PAGE,
         scope: 'europeana',
         fl: 'skos_prefLabel.*,isShownBy,isShownBy.thumbnail'
diff --git a/plugins/europeana/entity.js b/plugins/europeana/entity.js
index dac2fa4c..5ac821f3 100644
--- a/plugins/europeana/entity.js
+++ b/plugins/europeana/entity.js
@@ -58,7 +58,7 @@ export function getEntitySuggestions(text, params = {}) {
  * @param {string} type the type of the entity
  * @return {string} retrieved API name of type
  */
-function getEntityTypeApi(type) {
+export function getEntityTypeApi(type) {
   const names = {
     person: 'agent',
     topic: 'concept'
@@ -215,14 +215,14 @@ export function findEntities(entityUris) {
   if (entityUris.length === 0) return;
   const q = entityUris.join('" OR "');
   const params = {
-    query: `entity_uri:("${q}")`,
-    type: 'topic' /* needs to be dynamic */,
-    scope: 'europeana',
-    fl: 'skos_prefLabel.*,isShownBy,isShownBy.thumbnail'
+    query: `entity_uri:("${q}")`
   };
   return searchEntities(params)
     .then((response) => {
-      return response.entities;
+      return response.entities || [];
+    })
+    .catch((error) => {
+      throw error;
     });
 }
 
@@ -303,7 +303,6 @@ export function searchEntities(params = {}) {
   return axios.get(entityApiUrl('/search'), {
     params: {
       ...params,
-      type: getEntityTypeApi(params.type),
       wskey: config.entity.key
     }
   })
```

Reply:
thanks for your help with resolving this using getEntityTypeApi